### ISSUE TITLE
error message not showing in the UI

### ISSUE THREAD

**Issue by OmerKahani** [argoproj/argo-cd#1897]
**Describe the bug**
Using Helm chart the UI looks like: 
<img width="1679" alt="image" src="https://user-images.githubusercontent.com/8766193/60890783-b0e90000-a264-11e9-8d58-53172f82bcb9.png">
in the repo server I found:
`time="2019-07-09T13:04:08Z" level=error msg="`helm template . --name istio-init --namespace istio-system -f values.yaml` failed: Error: found in requirements.yaml, but missing in charts/ directory: istio-init\n"`
and in the application server:
`time="2019-07-09T13:13:01Z" level=warning msg="Error updating application: Application.argoproj.io \"istio-init\" is invalid:...error: code = Unknown desc = Error: no repository definition for https://storage.googleapis.com/istio-release/releases/1.1.5/charts. Please add them via 'helm repo add'... validation failure list:\nstatus.sync.revision in body is required" application=istio-init`

**To Reproduce**
1. add a new chart with a sub chart from helm repo that isn't in the argocd-cm

**Expected behavior**
The UI should show an error

**Version**
1.0.1

**Comment by alexec** [comment]
This should be fixed in v1.2.

**Comment by alexmt** [comment]
@OmerKahani , can you please attach app yaml if you still have it? I could not reproduce the error

**Comment by OmerKahani** [comment]
I have it from different error (folder not found in git repo)
`
apiVersion: argoproj.io/v1alpha1
kind: Application
  name: test-app
  namespace: argocd
spec:
  destination:
    namespace: default
    server: https://kubernetes.default.svc
  project: default
  source:
    helm:
      valueFiles:
      - values-production.yaml
    path: test-app
    repoURL: git@github.com:ORG/REPO.git
    targetRevision: master
` 

just change REPO to any repo, and apply the application

**Comment by alexmt** [comment]
This was fixed by https://github.com/argoproj/argo-cd/issues/1764. The fix is in 1.1. Quick workaround is to just disable CRD validation using 

```
kubectl patch customresourcedefinition applications.argoproj.io -p '[{ "op": "remove", "path": "/spec/validation" }]' --type=json
```
